main.cpp

```diff
@@ -131,233 +169,381 @@ struct CarWash
-    double SpeakerR = 5.0;
-    //    - mono audio (int)
+    double SpeakerLOutput = 5.0;
+    double SpeakerROutput = 5.0;
     int monoAudio = 0;
-    //3 things it can do:
-    //    - adjust volume    // Difference in initial volume setting and volume end value
+    
     double adjustVolume(double volumeKnobDiff);
-    //    - change input device
     void changeInputDevice();
-    //    - combine sound     // volume output mono-audio
     double combineSound(double SpeakerL, double SpeakerR);
+    void storeItemInArcadeBox(){}
+};
+
+double Speakers::combineSound(double SpeakerL, double SpeakerR)
+{
+    double combinedOutput = 0.0;
+    
+    if(monoAudio == 1) combinedOutput = 0.5 * (SpeakerL + SpeakerR);
+
+    return combinedOutput;
+}
+
+
+
+// 
+// UDT No. 9
+// 
+
+struct Coin 
+{
+    double coinDiameter = 0.0;
+    double coinWeight = 0.0;
+
+    bool inserted()
+    {
+        return true;
+    }
 };
 
 struct CoinBox 
 {
-    //5 properties:
-    //    - coin return (bool)
     bool coinReturn = false;
-    //    - amount of coin stored (int)
     int amtCoinStored = 134;
-    //    - coin slot size (float)
     double coinSlotSize = 0.958;
-    //    - amount of change backed-up (int)
     int amtBackedUpChange = 2;
-    //    - coin detector switch (bool)
     bool quarterDetected = true;
 
+    // 
+    // UDT No. 10
+    // 
+
     struct CoinSensor
     {
         bool isAQuarter = true;
         int amountInCents = 0;
-        float coinDiameter = 0.f;
-        float coinWeight = 0.f;
+        double coinDiameter = 0.00;
+        double coinWeight = 0.00;
         int creditToPlay = 3;
 
-        void inserted();
-        void acceptCoin(int country, bool isDirty = false);
-        int creditApplied(int amountPerCredit, int amountSinceLastGameFinished, int currentCredit);    // Updates credits based on amount of coin inserted
+        int detectTypeOfCoinInserted(double coinDiameterInches, double coinWeightGrams)
+        {
+            int coinType = 0;
+            
+            if((coinDiameterInches >= 0.74 || coinDiameterInches <= 0.76) && (coinWeightGrams >= 2.4 || coinWeightGrams <= 2.6))
+            {
+                coinType = 1;
+            }
+            else if((coinDiameterInches >= 0.695 || coinDiameterInches <= 0.715) && (coinWeightGrams >= 2.168 || coinWeightGrams <= 2.368))
+            {
+                coinType = 2;
+            }
+            else if((coinDiameterInches >= 0.825 || coinDiameterInches <= 0.845) && (coinWeightGrams >= 4.9 || coinWeightGrams <= 5.1))
+            {
+                coinType = 3;
+            }
+            else if((coinDiameterInches >= 0.945 || coinDiameterInches <= 0.965) && (coinWeightGrams >= 5.57 || coinWeightGrams <= 5.77))
+            {
+                coinType = 4;
+            }
+
+            return coinType;
+        }
+        bool acceptCoin(int coinType)
+        {
+            if(coinType == 0) return false;
+            else return true;
+        }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise